File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length ==0){
            return 0; 
        }
        

        HashSet <Integer> set = new HashSet<>(); 

        for(int i=0; i<nums.length; i++){
            set.add(nums[i]);
        }
        int longestSequence = 1; 
        for(int x: set){
            int num = x;
            int sequence = 1;
            if(!set.contains(x-1)){
                while(set.contains(num+1)){
                    sequence++; 
                    longestSequence = Math.max(longestSequence, sequence);
                    num = num+1;  
                }
            }

        }
            return longestSequence; 

    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return -1
        
        maxProfit = 0
        for i in range(len(prices)):
            for j in range(i+1, len(prices)):
                
                maxProfit = max(maxProfit, prices[j]-prices[i])
            
        
        return maxProfit